src/bitcoin/d/utils.rs

```diff
@@ -371,12 +371,13 @@ mod tests {
     #[test]
     fn bitcoind_roundup_progress() {
         assert_eq!(roundup_progress(0.6), 0.6);
-        assert_eq!(roundup_progress(0.67891), 0.678);
+        assert_eq!(roundup_progress(0.67891), 0.6789);
         assert_eq!(roundup_progress(0.98), 0.98);
         assert_eq!(roundup_progress(0.998), 0.998);
-        assert_eq!(roundup_progress(0.9476), 0.947);
-        assert_eq!(roundup_progress(0.998), 0.998);
-        assert_eq!(roundup_progress(0.9998), 1.0);
-        assert_eq!(roundup_progress(0.9991), 1.0);
+        assert_eq!(roundup_progress(0.9476), 0.9476);
+        assert_eq!(roundup_progress(0.94761), 0.9476);
```

Comment:
It would be good to include an additional test to show that numbers below 0.9999 are truncated rather than rounded up, e.g.
```diff
diff --git a/src/bitcoin/d/utils.rs b/src/bitcoin/d/utils.rs
index c317172b..ddd6d8af 100644
--- a/src/bitcoin/d/utils.rs
+++ b/src/bitcoin/d/utils.rs
@@ -376,6 +376,7 @@ mod tests {
         assert_eq!(roundup_progress(0.998), 0.998);
         assert_eq!(roundup_progress(0.9476), 0.9476);
         assert_eq!(roundup_progress(0.94761), 0.9476);
+        assert_eq!(roundup_progress(0.94769), 0.9476);
         assert_eq!(roundup_progress(0.9998), 0.9998);
         assert_eq!(roundup_progress(0.99998), 1.0);
         assert_eq!(roundup_progress(0.99991), 1.0);

```